File: daily/Sliding Window/209-Minimun-size-subarray-sum.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        n = len(nums)
        left = 0
        sum_num = 0
        res = float('inf')
        for right in range(n):
            sum_num += nums[right]

            while sum_num >= target:
                res = min(res, right - left + 1)
                sum_num -= nums[left]
                left += 1
        return res if res != float('inf') else 0

File: daily/Sliding Window/209-Minimun-size-subarray-sum.go
package daily

func minSubArrayLen(target int, nums []int) int {
    n := len(nums)
    left := 0
    current_sum := 0
    res := math.MaxInt32
    for right := 0; right < n; right++ {
        current_sum += nums[right]
        for current_sum <= target {
            res = min(res, right - left + 1)
            current_sum -= nums[left]
            left++
        }
    }
    
    if res == math.MaxInt32 {
        return 0
    }
    return res
}